model/project_matrix.go

```diff
@@ -247,6 +247,15 @@ func evaluateAxisTags(ase *axisSelectorEvaluator, axis string, selectors []strin
 	return out, errs
 }
 
+func GetVariantsWithMatrices(ase *axisSelectorEvaluator, axes []matrixAxis, bvs []parserBV) ([]parserBV, []error) {
+	if ase == nil {
```

Comment:
I think this `if` statement is unnecessary. `GetVariantsWithMatrices` is only called twice. In `TranslateProject` we construct an `ase` immediately before from `pp.Axes` and then pass it to the function. In `FindExpansionsForVariant` we pass `nil` but then also pass `pp.Axes`. I think you can remove the first argument to `GetVariantsWithMatrices` since you're always passing `pp.Axes`. You can then always construct the `ase` in `GetVariantsWithMatrices`. And in `TranslateProject` remove constructing the `ase` before calling `GetVariantsWithMatrices`.

Reply:
We do need ase in TranslateProject for the NewVariantSelectorEvaluator. I wanted to save some complexity in TranslateProject by passing it through since it's an expensive function; I can change to just always passing it in instead though if you prefer.

Reply:
Oh I missed that. That makes sense, thanks. LGTM.